vendor_pricelist_disable/tests/test_purchase_pricelist_disable.py

```diff
@@ -0,0 +1,192 @@
+from odoo.tests import tagged
+from odoo.tests.common import TransactionCase
+
+
+@tagged("post_install", "-at_install")
+class TestPurchasePricelistDisable(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Test Product",
+                "type": "product",
+            }
+        )
+
+        cls.partner1 = cls.env["res.partner"].create(
+            {
+                "name": "Vendor with auto pricelist",
+            }
+        )
+
+        cls.partner2 = cls.env["res.partner"].create(
+            {
+                "name": "Vendor without auto pricelist",
+            }
+        )
+
+        cls.company2 = cls.env["res.company"].create(
+            {
+                "name": "Test Company",
+            }
+        )
+
+    def test_01_pricelist_enabled(self):
+        """Test supplier pricelist creation when enabled"""
+        self.env.company.purchase_pricelist_disable_autocreate = True
+
+        po = self.env["purchase.order"].create(
```

Comment:
You can create the purchase order and the purchase order line in the setUpClass method, so you don't duplicate code in test_01_pricelist_enabled and test_02_pricelist_disabled